ros_ign_gazebo_demos/package.xml

```diff
@@ -24,6 +24,7 @@
   <exec_depend>rqt_plot</exec_depend>
   <exec_depend>rqt_topic</exec_depend>
   <exec_depend>rviz2</exec_depend>
+  <exec_depend condition="$IGNITION_VERSION != garden">>sdformat_urdf</exec_depend>
```

Comment:
Should this be

```suggestion
  <exec_depend condition="$IGNITION_VERSION == garden">sdformat_urdf</exec_depend>
```

Reply:
Removed the extra > but the condition is correct, since we don't want to install from binaries if its garden

Reply:
Oh I see what you're saying. The `package.xml` file isn't used just by `rosdep` to install binary dependencies though. It's also used by `colcon` and `bloom`.

Excluding Garden here gives the impression that the demos that use `sdformat_urdf` can't be used with Garden, which is not true. So we should be able to leave this condition out. CI is running `rosdep` with the `-r` flag, so it won't fail when it can't find this dependency.

Reply:
CI was failing without it though
```
ERROR: the following rosdeps failed to install
  apt: command [apt-get install -y ros-rolling-sdformat-urdf] failed
  apt: Failed to detect successful installation of [ros-rolling-sdformat-urdf]
```

Reply:
It looks like the issue here is a bit higher:

```
gz-tools2 : Breaks: ignition-tools but 1.5.0-1~jammy is to be installed
```

We shouldn't be pulling `ignition-tools` on the Garden job though, only `gz-tools`. Not sure yet what's pulling that.

Reply:
`libsdformat12-dev` is pulling that, see this comment https://github.com/gazebo-tooling/release-tools/issues/529#issuecomment-1231016550

Reply:
Ok, so instead of trying to suppress the installation of `sdformat_urdf` here, I propose we do it just for CI at `build-and-test.sh`. Something like this (untested):

```diff
diff --git a/.github/workflows/build-and-test.sh b/.github/workflows/build-and-test.sh
index 1ac4206..aa516d1 100755
--- a/.github/workflows/build-and-test.sh
+++ b/.github/workflows/build-and-test.sh
@@ -17,6 +17,8 @@ if [ "$GZ_VERSION" == "garden" ]; then
   wget https://packages.osrfoundation.org/gazebo.key -O - | apt-key add -
 
   GZ_DEPS="libgz-sim7-dev"
+
+  ROSDEP_ARGS="--skip-keys='sdformat-urdf'"
 fi
 
 # Fortress comes through rosdep for Focal and Jammy
@@ -31,7 +33,7 @@ apt-get install -y $GZ_DEPS \
 
 rosdep init
 rosdep update
-rosdep install --from-paths ./ -i -y -r --rosdistro $ROS_DISTRO
+rosdep install --from-paths ./ -i -y -r --rosdistro $ROS_DISTRO $ROSDEP_ARGS
 
 # Build.
 source /opt/ros/$ROS_DISTRO/setup.bash
```

This way colcon users would still get that dependency.

Reply:
Seems to have worked, CI is green!

Reply:
Awesome! We can remove the condition here then! This way, folks building with colcon will get the dependencies resolved correctly

```
<exec_depend>sdformat_urdf</exec_depend>
```

Reply:
All set